idx/memory/memory.go

```diff
@@ -552,13 +553,26 @@ func (m *MemoryIdx) resolveIDs(ids TagIDs) []string {
 			continue
 		}
 
-		res[i] = def.NameWithTags()
+		name := def.NameWithTags()
+		node, ok := tree.Items[name]
+		if !ok {
+			corruptIndex.Inc()
+			log.Error(3, "memory-idx: node %s missing. Index is corrupt.", name)
+			continue
+		}
```

Comment:
note that because we went with the "set len up front, then fill in `res[i]` as we go" approach, in this case, as well as the above corruptIndex case, we will have an uninitialized slot in the slice for pos `i` which seems like it would probably break stuff. so instead, let's `res := make([]idx.Node, 0, len(ids))` and then appendthe valid entries

Reply:
3bdadaf